pkg/component/component.go

```diff
@@ -175,34 +175,40 @@ func ListAllClusterComponents(client kclient.ClientInterface, namespace string)
 		}
 
 		// Generate the appropriate "component" with all necessary information
-		component := OdoComponent{
+		component := api.ComponentAbstract{
 			Name:      name,
 			ManagedBy: managedBy,
 			Type:      componentType,
 		}
 		mode := odolabels.GetMode(labels)
-		found := false
+		componentFound := false
 		for v, otherCompo := range components {
 			if component.Name == otherCompo.Name {
-				found = true
+				componentFound = true
 				if mode != "" {
-					components[v].Modes[mode] = true
+					modeFound := false
+					for _, m := range components[v].RunningIn {
+						if m == api.RunningMode(mode) {
+							modeFound = true
+							break
+						}
+					}
+					if !modeFound {
+						components[v].RunningIn = append(components[v].RunningIn, api.RunningMode(mode))
+						sort.Sort(components[v].RunningIn)
+					}
```

Comment:
I couldn't determine if integration tests covered this. Would it be possible to update the unit test cases to make sure the changes here are covered?